### DISCUSSION TITLE
streamlit 서버 접속 오류

### DISCUSSION THREAD

**Discussion by Heo-Gyeom** [WHATDO-TEAM/whatdo#56] [트러블슈팅]
<img width="558" alt="스크린샷 2024-11-04 오전 10 34 44" src="https://github.com/user-attachments/assets/2a90f105-87e1-4459-ba05-14387066650f">

    floors = st.number_input("층수 선택", min_value=1, max_value=10, value=1)
    visitor_cnt = st.number_input("방문객 수 선택", min_value=1, max_value=10, value=1)
    
    # 날짜 범위 선택
    start_date = st.date_input("시작 날짜")
    end_date = st.date_input("종료 날짜")

    if st.button("데이터 분석"):
        floors_str = ','.join(map(int, floors))
        params = {
            "room_floor_seq": floors_str,
            "visitor_cnt": visitor_cnt,
            "start_date": start_date.strftime("%Y-%m-%d"),
            "end_date": end_date.strftime("%Y-%m-%d")
        }

`floors_str = ','.join(map(int, floors))`: floor을 int 타입으로 바꾸기
(visitor_cnt도 int 타입인데 이 코드가 필요없어 보여 삭제 후 진행)

<img width="568" alt="스크린샷 2024-11-04 오전 10 46 58" src="https://github.com/user-attachments/assets/0330c7a5-e018-4015-a36c-41b5992cb2b5">

서버에 접속이 안됨 

    @app.get("/filter_data")
    async def filter_data(floors: str, visitor_cnt: int, start_date: str, end_date: str):
        floor_list = list(map(int, floors.split(',')))

     filtered_data = df[
        (df['floor'].isin(floor_list)) &
        (df['visitor_cnt'] == visitor_cnt) &
        (df['check_in'] >= start_date) &
        (df['check_in'] <= end_date)
    ]
        
 'floors' csv에 데이터 컬럼값 오류 -> room_floor_seq 변경 (int 값으로 변경)
` floor_list = list(map(int, floors.split(',')))` 
여러개의 층을 한번에 입력 가능하도록 하는 코드 -> 한개의 층만 출력 (코드 삭제)

     filtered_data = df[
        (df['floor'].isin(floor_list)) &
        (df['visitor_cnt'] == visitor_cnt) &
        (df['check_in'] >= start_date) &
        (df['check_in'] <= end_date)
    ]

 (df['floor'].isin(floor_list)) & -> isin 함수 삭제 
 
 #수정 코드 

    @app.get("/filter_data")
    async def filter_data(room_floor_seq: int, visitor_cnt: int, start_date: str, end_date: str):
     
    # 날짜 필터링을 위해 문자열을 날짜 형식으로 변환
    start_date = pd.to_datetime(start_date)
    end_date = pd.to_datetime(end_date)
  
    # 조건에 맞는 데이터 필터링
    filtered_data = df[
        (df['room_floor_seq'] == room_floor_seq) &
        (df['visitor_cnt'] == visitor_cnt) &
        (df['check_in'] >= start_date) &
        (df['check_in'] <= end_date)
    ]

<img width="835" alt="스크린샷 2024-11-04 오전 11 15 32" src="https://github.com/user-attachments/assets/6c9ed714-60d2-4ed5-bbf4-e592f0643651">

**Comment by Kdreamtomaster** [comment]
방문객이 없는 날짜의 경우에는 0으로 출력되야 정확한 추이를 볼 수 있을것 같아요
